### DISCUSSION TITLE
[Bug]: 502 bad gateway on fresh install

### DISCUSSION THREAD

**Discussion by LarryBarker** [beyondcode/herd-community#501] [Q&A]
### Operating system version

macOS 

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.5.0 (23)

### PHP Version

Any

### Bug description

After a fresh install, visiting a site in the browser produces a 502 bad gateway error. FPM logs are below. This is a corporate laptop that uses LDAP to connect to corporate directory services, and likely the issue. We have tried force stopping services, enabling root user in macOS, switching PHP versions, etc. but continue to face the same error.

We have confirmed /etc/sudoers.d/herd exists and has the appropriate line: 

```
trevsta@bg-al04 ~ % cd /etc/sudoers.d
trevsta@bg-al04 sudoers.d % ls -a
.	..	herd
trevsta@bg-al04 sudoers.d % cat herd
%admin ALL=(root) NOPASSWD:SETENV: /Applications/Herd.app/Contents/Resources/*, /usr/bin/security, /bin/kill
trevsta@bg-al04 sudoers.d %
```

It seems like there is an ownership issue with Herds application support files. See below:

```
trevsta@bg-al04 Herd % ls -la
total 192
drwxr-xr-x@ 14 trevsta CORP\Domain Users  448 Mar 13 21:07 .
drwx------+ 40 trevsta CORP\Domain Users  1280 Mar 13 20:11 ..
drwxr-xr-x@ 4 trevsta CORP\Domain Users  128 Mar 13 21:07 .pid
-rw-r--r--@ 1 trevsta CORP\Domain Users 65536 Mar 13 19:58 HerdCoreData.sqlite
-rw-r--r--@ 1 trevsta CORP\Domain Users 32768 Mar 13 21:07 HerdCoreData.sqlite-shm
-rw-r--r--@ 1 trevsta CORP\Domain Users   0 Mar 13 19:58 HerdCoreData.sqlite-wal
drwxr-xr-x@ 4 trevsta CORP\Domain Users  128 Mar 13 19:58 Log
drwxr-xr-x@ 20 trevsta CORP\Domain Users  640 Mar 13 21:07 bin
drwxr-xr-x@ 7 trevsta CORP\Domain Users  224 Mar 13 19:58 config
lrwxr-xr-x  1 trevsta CORP\Domain Users   65 Mar 13 21:07 herd-debug.sock -> /Users/trevsta/Library/Application Support/Herd/herd-debug83.sock
lrwxr-xr-x@ 1 trevsta CORP\Domain Users   59 Mar 13 21:06 herd.sock -> /Users/trevsta/Library/Application Support/Herd/herd83.sock
srwxrwxrwx@ 1 trevsta CORP\Domain Users   0 Mar 13 21:05 herd74.sock
srwxrwxrwx@ 1 trevsta CORP\Domain Users   0 Mar 13 20:01 herd80.sock
srwxrwxrwx  1 trevsta CORP\Domain Users   0 Mar 13 21:07 herd83.sock
trevsta@bg-al04 Herd %
[9:22](https://bgprod.slack.com/archives/C06PSAFL341/p1710382922433239)
/Users/trevsta/Library/Application Support/Herd
trevsta@bg-al04 Herd %
```

<img width="1406" alt="image" src="https://github.com/beyondcode/herd-community/assets/28734844/58ea093a-27e3-4eba-96de-a8ea69677551">

Based on that it looks like Herd is owned by the `CORP\Domain Users` group? 

### Steps to reproduce

1. Download and install Herd
2. Set up a new Laravel project (or clone an existing) in the appropriate directory(ies)
3. Visit project in browser

### Relevant log output

```shell
[13-Mar-2024 20:40:21] NOTICE: [pool herd] 'user' directive is ignored when FPM is not running as root
[13-Mar-2024 20:40:21] NOTICE: [pool herd] 'group' directive is ignored when FPM is not running as root
[13-Mar-2024 20:40:21] ERROR: [pool herd] failed to chown() the socket '/Users/trevsta/Library/Application Support/Herd/herd74.sock': Operation not permitted (1)
[13-Mar-2024 20:40:21] ERROR: FPM initialization failed
```

**Comment by dwabuu** [comment]
Did you try isolating the PHP version? This usually works for me.

**Comment by GuyVervilleSpiria** [comment]
I have the same problem here on my MacBookPro. At home, on my MacBookAir (more recent), I had not problem. I can't see no log either... Trying to secure a site (even a simple index.php with only a php command) doesn't work. Trying to change the PHP version doesn't work either. I uninstalled and reinstalled Herd. I have a pro license.

**Comment by GuyVervilleSpiria** [comment]
I have somewhat identified the cause. I had to allow Herd in the security settings. http links work, but not the https.

**Comment by LarryBarker** [comment]
> Did you try isolating the PHP version? This usually works for me.

@dwabuu Can you elaborate here? How do we isolate the PHP version? Also not sure this will help here because this issue seems to be more permission related somehow. 

Follow up, we have already isolated PHP for this site
<img width="1041" alt="image" src="https://github.com/beyondcode/herd-community/assets/28734844/126526d0-9af5-47a8-8e2f-80346d104c6b">





> I have somewhat identified the cause. I had to allow Herd in the security settings. http links work, but not the https.

@GuyVervilleSpiria Where would we allow Herd in security settings?

**Comment by GuyVervilleSpiria** [comment]
@LarryBarker I allowed access to read/write. But I realized another thing. I can't have colima/ddev operating at the same time as Herd. When firing Herd, I lose access to my nginx engine in ddev (or ddev is killing the Herd nginx)...

**Comment by LarryBarker** [comment]
@GuyVervilleSpiria still not following. You mean on macOS, system preferences > privacy and security?


Interestingly enough, I just looked at my FPM logs on my personal laptop and I have the same error so it's not just related to a corporate device with LDAP.

**Comment by GuyVervilleSpiria** [comment]
@LarryBarker Yes, it's on macOS system preferences, privacy and security. But my main point is that Herd seems not "isolated", hence my comment about ddev (https://ddev.com/)  and herd coexisting.
